src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -5,26 +5,30 @@ class ConfigurationHelper implements Serializable {
 
     def static SEPARATOR = '/'
 
+    static ConfigurationHelper newInstance(Script workflow, Script step, Map config = [:]) {
+        new ConfigurationHelper(workflow, step, config)
+    }
+
     static ConfigurationHelper newInstance(Script step, Map config = [:]) {
-        new ConfigurationHelper(step, config)
+        new ConfigurationHelper(null, step, config)
     }
 
     ConfigurationHelper loadStepDefaults(Map compatibleParameters = [:]) {
-        DefaultValueCache.prepare(step)
-        this.config = ConfigurationLoader.defaultGeneralConfiguration()
-        mixin(ConfigurationLoader.defaultGeneralConfiguration(), null, compatibleParameters)
+        DefaultValueCache.prepare(workflow)
+        this.config = ConfigurationLoader.defaultGeneralConfiguration(workflow)
+        mixin(ConfigurationLoader.defaultGeneralConfiguration(workflow), null, compatibleParameters)
         mixin(ConfigurationLoader.defaultStepConfiguration(null, name), null, compatibleParameters)
     }
 
     private Map config
-    private Script step
+    private Script workflow
     private String name
     private Map validationResults = null
     private String dependingOn
 
-    private ConfigurationHelper(Script step, Map config){
+    private ConfigurationHelper(Script workflow, Script step, Map config){
         this.config = config ?: [:]
-        this.step = step
+        this.workflow = workflow?:step
```

Comment:
I think this is dangerous as step and workflow are different and it could happen that in some steps workflow is taken and in others step which might be difficult to debug.

Reply:
Yes, how can we solve that? As I see it, step is only there for the STEP_NAME variable, maybe we should change the signature to `private ConfigurationHelper(Script workflow, String stepName, Map config)`